mmcv/transforms/loading.py

```diff
@@ -0,0 +1,270 @@
+    ) -> None:
+        self.to_float32 = to_float32
+        self.color_type = color_type
+        self.imdecode_backend = imdecode_backend
+        self.file_client_args = file_client_args.copy()
+        self.file_client = None
+
+    def transform(self, results: dict) -> dict:
+        """Functions to load image.
+
+        Args:
+            results (dict): Result dict from :obj:`mmdet.CustomDataset`.
+
+        Returns:
+            dict: The dict contains loaded image and meta information.
+        """
+
+        if self.file_client is None:
+            self.file_client = mmcv.FileClient(**self.file_client_args)
+
+        filename = results['img_path']
+        img_bytes = self.file_client.get(filename)
+        img = mmcv.imfrombytes(
+            img_bytes, flag=self.color_type, backend=self.imdecode_backend)
+        if self.to_float32:
+            img = img.astype(np.float32)
+
+        results['img'] = img
+        height, width = img.shape[:2]
+        results['height'] = height
+        results['width'] = width
+        results['ori_height'] = height
+        results['ori_width'] = width
+        return results
+
+    def __repr__(self):
+        repr_str = (f'{self.__class__.__name__}('
+                    f'to_float32={self.to_float32}, '
+                    f"color_type='{self.color_type}', "
+                    f"imdecode_backend='{self.imdecode_backend}', "
+                    f'file_client_args={self.file_client_args})')
+        return repr_str
+
+
+class LoadAnnotation(BaseTransform):
+    """ Load and process the `instances` and `seg_map` annotation provided by
+    dataset. `LoadAnnotation` loads only one image annotation.
+
+    The annotation format is as the following:
+
+    ```python
+    {
+        'instances':
+        [
+            {
+            # List of 4 numbers representing the bounding box of the
+            # instance, in (x1, y1, x2, y2) order.
+            'bbox': [x1, y1, x2, y2],
+
+            # Label of image classification.
+            'bbox_label': 1,
+
+            # Used in key point detection.
+            # Can only load the format of [x1, y1, v1,…, xn, yn, vn]. v[i]
+            # means the visibility of this keypoint. n must be equal to the
+            # number of keypoint categories.
+            'keypoints': [x1, y1, v1, ..., xn, yn, vn]
+            }
+        ]
+        # Filename of semantic or panoptic segmentation ground truth file.
+        'seg_map': 'a/b/c'
+    }
+    ```
+    After load and process:
+
+    ```python
+    {
+        'gt_bboxes': np.ndarray(N, 4) # In (x1, y1, x2, y2) order, float type.
+        'gt_bboxes_labels': np.ndarray(N, ) # In int type.
+        'gt_semantic_seg': np.ndarray (H, W) # In uint8 type.
+        'gt_keypoints': np.ndarray(N, NK, 3) # in (x, y, v) order, float type.
+
+    }
+    ```
+
+    Required Key:
+
+    - instances
+        - bbox (optional)
+        - bbox_label (optional)
+        - keypoints
+    - seg_map
```

Comment:
The reStructured syntax is indent-sensitive and blank-line-sensitive. For nested list, please use:
```suggestion
    - instances
    
      - bbox (optional)
      - bbox_label (optional)
      - keypoints
        
    - seg_map
```